sdk/metric/view/view.go

```diff
@@ -17,22 +17,186 @@
+	return v.instrumentName == nil || v.instrumentName.MatchString(name)
+}
+
+func (v View) matchLibraryName(name string) bool {
+	return v.scope.Name == "" || name == v.scope.Name
+}
+
+func (v View) matchLibraryVersion(version string) bool {
+	return v.scope.Version == "" || version == v.scope.Version
+}
+
+func (v View) matchLibrarySchemaURL(schemaURL string) bool {
+	return v.scope.SchemaURL == "" || schemaURL == v.scope.SchemaURL
 }
 
-// TODO (#2837): add getter functions for all the internal fields of a Config.
+func (v View) match(i Instrument) bool {
+	return v.matchName(i.Name) &&
+		v.matchLibraryName(i.Scope.Name) &&
+		v.matchLibrarySchemaURL(i.Scope.SchemaURL) &&
+		v.matchLibraryVersion(i.Scope.Version)
+}
 
-// Option applies a configuration option value to a view Config.
+// Option applies a Configuration option value to a View. All options
+// will be used together to determine match and transforms.
 type Option interface {
-	apply(Config) Config
+	apply(View) View
 }
 
-// TODO (#2837): implement view match options.
-// TODO (#2837): implement view annotation options.
+type optionFunc func(View) View
+
+func (f optionFunc) apply(v View) View {
+	return f(v)
+}
 
-// New returns a new and configured view Config.
-func New(opts ...Option) Config {
-	return Config{}
+// MatchInstrumentName will match an instrument based on the its name.
+// This will accept wildcards of * for zero or more characters, and ? for
+// exactly one character. A name of "*" (default) will match all instruments.
+func MatchInstrumentName(name string) Option {
+	return optionFunc(func(v View) View {
+		if strings.ContainsAny(name, "*?") {
+			v.hasWildcard = true
+		}
+		name = regexp.QuoteMeta(name)
+		name = "^" + name + "$"
+		name = strings.ReplaceAll(name, "\\?", ".")
+		name = strings.ReplaceAll(name, "\\*", ".*")
+		v.instrumentName = regexp.MustCompile(name)
+		return v
+	})
 }
+
+// TODO (#2813): Implement MatchInstrumentKind when InstrumentKind is defined.
+// TODO (#2813): Implement MatchNumberKind when NumberKind is defined.
+
+// MatchInstrumentationLibrary will do an exact match on any
+// instrumentation.Scope field that is non-empty (""). The default is to match all
+// instrumentation scopes.
+func MatchInstrumentationLibrary(lib instrumentation.Library) Option {
+	return optionFunc(func(v View) View {
+		v.scope = lib
+		return v
+	})
+}
+
+// WithName will rename the instrument the view matches. If not used or empty the
+// instrument name will not be changed. Must be used with a non-wildcard
+// instrument name match. The default does not change the instrument name.
+func WithName(name string) Option {
+	return optionFunc(func(v View) View {
+		v.name = name
+		return v
+	})
+}
+
+// WithDescription will change the description of the instruments the view
+// matches to desc. If not used or empty the description will not be changed.
+func WithDescription(desc string) Option {
+	return optionFunc(func(v View) View {
+		v.description = desc
```

Comment:
Based on the option comment, should this do nothing if `desc` is empty?

```suggestion
		if desc != "" {
			v.description = desc
		}
```

Reply:
The logic for that is in the TransformDescription.

What _should_ happen if someone uses:
```go
WithDescription("Some New Description"),
WithDescription(""),
```